### PR TITLE
Fix usability issues with BlotVector.

### PR DESCRIPTION
**Pull request by atrick**
- Add const getItems().

- Fix const find().

- erase() returns a boolean.

- Set erase() should not perform two lookups.

The implementation is covered by the unit tests with a small addition.

Other trivial API changes are trivially tested in upcoming commits.

### CODE REVIEW

**Comment by atrick** [comment]
@swift-ci smoke test.

**Comment by atrick** [comment]
@gottesmm can you take a look? Just some very minor improvements that I needed.

**Comment by atrick** [comment]
@swift-ci smoke test OS X Platform.

File: include/swift/Basic/BlotMapVector.h [hunk 1]
```diff
@@ -85,27 +88,34 @@ class BlotMapVector {
...
     /// intact, but clients must be prepared for zeroed-out keys when iterating.
+    ///
+    /// Return true if the element was erased.
+    bool erase(const KeyT &Key) { return blot(Key); }
+
+    /// Eliminate the element at the given iterator. Instead of removing the
+    /// element from the vector, just zero out the key in the vector. This
+    /// leaves iterators intact, but clients must be prepared for zeroed-out
+    /// keys when iterating.
     void erase(iterator I) { erase((*I)->first); }
```
Comment (gottesmm) [inline include/swift/Basic/BlotMapVector.h]: Why doesn't this also return a bool?

File: include/swift/Basic/BlotMapVector.h [hunk 2]
```diff
@@ -85,27 +88,34 @@ class BlotMapVector {
...
 
-    /// This is similar to erase, but instead of removing the element from the
+    /// Eliminate the element at `Key`. Instead of removing the element from the
     /// vector, it just zeros out the key in the vector. This leaves iterators
     /// intact, but clients must be prepared for zeroed-out keys when iterating.
-    void blot(const KeyT &Key) {
+    ///
+    /// Return true if the element was found and erased.
+    bool blot(const KeyT &Key) {
```
Comment (gottesmm) [inline include/swift/Basic/BlotMapVector.h]: Maybe just move this into erase? I don't remember why it is separate.

**Review by gottesmm — COMMENTED** [review]
Two questions.

File: include/swift/Basic/BlotMapVector.h (hunk 2 above)
Comment (atrick) [reply include/swift/Basic/BlotMapVector.h]: I don't understand why it's separate but wasn't terribly interested in changing all the clients in this PR.

File: include/swift/Basic/BlotMapVector.h (hunk 1 above)
Comment (atrick) [reply include/swift/Basic/BlotMapVector.h]: Presumably you're passing a valid iterator, so wouldn't make much sense.

File: include/swift/Basic/BlotMapVector.h (hunk 2 above)
Comment (gottesmm) [reply include/swift/Basic/BlotMapVector.h]: Ok. If it isn't too much work, can you do it in a follow on commit? Or I may do it in a drive by change later.

File: include/swift/Basic/BlotMapVector.h (hunk 1 above)
Comment (gottesmm) [reply include/swift/Basic/BlotMapVector.h]: ok.

**Comment by gottesmm** [comment]
I am going to merge this and do some cleanups afterwards since @atrick is on vacation.

**Comment by atrick** [comment]
Thanks @gottesmm

**Pull request merged**